jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import numpy.ma as ma
+
+from traitlets import List, Unicode, Bool, Int, observe
+
+from astropy.table import QTable
+from astropy.coordinates import SkyCoord
+from astroquery.sdss import SDSS
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+
+__all__ = ['Catalogs']
+
+
+@tray_registry('imviz-catalogs', label="Imviz Catalogs")
+class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
+    template_file = __file__, "catalogs.vue"
+    catalog_items = List([]).tag(sync=True)
+    catalog_selected = Unicode("").tag(sync=True)
+    results_available = Bool(False).tag(sync=True)
+    number_of_results = Int(0).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.catalog_items = ["SDSS"]
+
+        if self.catalog_selected not in self.catalog_items:
+            self.catalog_selected = self.catalog_items[0]
+
+    @observe("catalog_selected")
+    def vue_do_search(self, *args, **kwargs):
+        # no querying occurs while the plugin has not been opened
+        if not self.plugin_opened:
+            return
```

Comment:
I don't think you need this, since the plugin has to be open to click the button, and maybe someone would want to call this method manually from the API to show the markers?

If you want to make the user-API a little more convenient, consider a `def search(self)` that contains all of this logic and then have `def vue_do_search(self, *args, **kwargs)` only make a call to `self.search()`.  This way the user-facing API (`plugin.search()`) doesn't have the vue-prefix and doesn't accept meaningless inputs in `*args, **kwargs`.  Feel free to defer to follow-up effort though!

Reply:
I removed that `if` statement and added the `def search(self)` function! I agree that these changes make the user-API a little more convenient.